account_default_draft_move forward port

laetitia-gangloff opened the pull request:
In certain case (automatic invoice) we doesn't need to check generate entries and we want to post them.
So I introduce a new account config parameter : use_journal_setting to use the setting 'entry_posted' of the journal.

Forward port of #148 
## 
#205 :

The goal is to use update_posted set on journal to display the cancel button on account_move.

As suggest by @sbidoul in #150, I "make the cancel button appear only if update_posted is true and account_cancel is installed".

Forward port of #175

sbidoul commented:
@laetitia-gangloff Can you also extract the description in a README.rst according to the OCA template: https://github.com/OCA/maintainer-tools/tree/master/template/module

I suggest the following as description reflecting the new features:

In Odoo there is a flag "Skip draft moves" on accounting journals that controls if newly created account move stay in draft state or are posted immediately. This flag is honored throughout the system except for invoices and bank statements, for which moves are posted as soon as the documents are validated. Therefore, to correct errors on invoices and bank statement after their validation, one must install the account_cancel module, which in turn potentially enables the cancellation of all account moves. This is dangerous.

This module does the following:
- It makes sure the flag "Skip draft moves" is always honored, including when validating invoices and bank statements [1]. This enables different workflows, such as 1/ the automatic posting of moves coming from some journals (such as invoices coming from e-commerce), or 2/ moves remaining in draft state, letting accountants validate them and correct the corresponding invoices before posting the moves.
- It hides the Cancel button on account moves unless account_cancel is installed and cancellation is allowed on the corresponding journal.

Additionally, the Post button is hidden on account moves, therefore providing a framework where the user work with draft moves until everything is correct, and validates all account moves at once and the end. [2]

[1] provided the parameter is set in Settings > Configuration, if this parameter is not set the moves generated from invoices and bank statement remain in draft state. This parameter is not set by default for backward compatibility reasons.
[2] this behaviour will be made configurable in the near future.

cc/ @fclementic2c

Commit:
Add a configuration parameter to use entry_posted journal setting. Add update_posted in journal items to display or not the button cancel.

sbidoul commented:
:+1: when travis is green

sbidoul committed:
[IMP] account_default_draft_move: improve README

The new text should better reflect the recent evolutions of the module
and the various workflows you can setup with it.

adrienpeiffer commented:
:+1:

sbidoul committed:
[IMP] account_default_draft_move: add runbot button and roadmap section

sbidoul commented:
I rebased and improved the README a bit. I took the liberty to add ACSONE in the authors, given the amount of thinking and code we put into this module.

IMO this PR is good for merging.

lmignon commented:
:+1:

fclementic2c commented:
:+1:
